sdk/rt/speechmatics/rt/_async_multi_channel_client.py

```diff
@@ -0,0 +1,340 @@
+        except Exception as e:
+            self._logger.error("Audio producer error: %s", e)
+            self._session_done_evt.set()
+
+    def _on_recognition_started(self, msg: dict[str, Any]) -> None:
+        """Handle RecognitionStarted message from server."""
+        self._session.session_id = msg.get("id")
+        self._rec_started_evt.set()
+        self._logger.debug(
+            "Recognition started (session_id=%s)",
+            self._session.session_id,
+        )
+
+    def _on_eot(self, msg: dict[str, Any]) -> None:
+        """Handle EndOfTranscript message from server."""
+        self._session_done_evt.set()
+
+    def _on_error(self, msg: dict[str, Any]) -> None:
+        """Handle Error message from server."""
+        self._session_done_evt.set()
+        raise TranscriptionError(msg.get("reason", "unknown"))
+
+    def _on_warning(self, msg: dict[str, Any]) -> None:
+        """Handle Warning message from server."""
+        self._logger.warning("Server warning: %s", msg.get("reason", "unknown"))
+
+    async def close(self) -> None:
+        """
+        Close the client and clean up resources.
+
+        Ensures the session is marked as complete and delegates to the base
+        class for full cleanup including WebSocket connection termination.
+        """
+        self._session_done_evt.set()
+        await super().close()
+
+    def _validate_diarization_config(
+        self, transcription_config: TranscriptionConfig, sources: dict[str, BinaryIO]
+    ) -> None:
+        """
+        Validate diarization settings and source-channel consistency.
+
+        Raises:
+            ConfigurationError: If any part of the configuration is invalid.
+        """
+        if not sources:
+            raise ConfigurationError("Audio input sources cannot be empty.")
+
+        dz_options = ["channel", "channel_and_speaker"]
+        if transcription_config.diarization and transcription_config.diarization not in dz_options:
+            raise ConfigurationError(
+                f"Diarization must be one of {dz_options}, got '{transcription_config.diarization}'."
+            )
+
+        chan_labels = transcription_config.channel_diarization_labels
+        if chan_labels:
+            if len(chan_labels) != len(sources):
+                raise ConfigurationError(f"Expected {len(sources)} channel labels, got {len(chan_labels)}.")
+
+            if len(set(chan_labels)) != len(chan_labels):
+                raise ConfigurationError("Channel labels must be unique.")
+
+    def _remap_sources_with_labels(
+        self, transcription_config: TranscriptionConfig, sources: dict[str, BinaryIO]
+    ) -> dict[str, BinaryIO]:
+        """
+        Set default diarization values and return sources remapped to channel labels.
+
+        Returns:
+            A dict mapping channel labels to audio sources.
+        """
+        if transcription_config.diarization is None:
+            transcription_config.diarization = "channel"
```

Comment:
I wonder if this should happen in the diarization validation function instead? To me it was non-obvious this happens inside a function called _remap_sources_with_labels so I spent a bit of time trying to find where this assignement happened in the code